cpp/src/io/parquet/page_enc.cu

```diff
@@ -2109,22 +2115,17 @@ void InitPageFragments(PageFragment *frag,
  * @param[out] groups Statistics groups [num_columns x num_fragments]
  * @param[in] fragments Page fragments [num_columns x num_fragments]
  * @param[in] col_desc Column description [num_columns]
- * @param[in] num_fragments Number of fragments
- * @param[in] num_columns Number of columns
- * @param[in] fragment_size Max size of each fragment in rows
  * @param[in] stream CUDA stream to use, default 0
  */
-void InitFragmentStatistics(statistics_group *groups,
-                            const PageFragment *fragments,
-                            const parquet_column_device_view *col_desc,
-                            int32_t num_fragments,
-                            int32_t num_columns,
-                            uint32_t fragment_size,
+void InitFragmentStatistics(device_2dspan<statistics_group> groups,
+                            device_2dspan<PageFragment const> fragments,
+                            device_span<parquet_column_device_view const> col_desc,
                             rmm::cuda_stream_view stream)
 {
-  dim3 dim_grid(num_columns, (num_fragments + 3) >> 2);  // 1 warp per fragment
-  gpuInitFragmentStats<<<dim_grid, 128, 0, stream.value()>>>(
-    groups, fragments, col_desc, num_fragments, num_columns, fragment_size);
+  auto num_columns              = col_desc.size();
+  auto num_fragments_per_column = fragments.size().second;
+  dim3 dim_grid(num_columns, (num_fragments_per_column + 3) >> 2);  // 1 warp per fragment
```

Comment:
Using `div_rounding_up_unsafe(num_fragments_per_column, 128/cudf::detail::warp_size)` clarifies the grid dimension logic a bit better here.